crates/torii/core/src/engine.rs

```diff
@@ -165,34 +156,28 @@ impl<'db, P: Provider + Sync + Send> Engine<'db, P> {
         Ok(())
     }
 
-    async fn process(&mut self, block: MaybePendingBlockWithTxs) -> Result<()> {
-        let block: BlockWithTxs = match block {
-            MaybePendingBlockWithTxs::Block(block) => block,
-            _ => return Ok(()),
-        };
-
-        Self::process_block(self, &block).await?;
+    async fn process(&mut self, event: EmittedEvent, current_block: &mut u64) -> Result<()> {
+        if event.block_number.unwrap() > *current_block {
```

Comment:
we unwrap this a lot. should we do it safely and remove some of the repetition below?